[RNMobile] Ensure font is scaled on iOS when accessibility settings are changed

SiobhyB opened the pull request:
> **Note** There are times where the font appears overly large when used with custom fonts and block-based themes, this will be addressed in a separate Gutenberg PR for _both_ platforms. The scope of this specific PR is to enable support for Dynamic Type on iOS.

## Related PRs

* `Gutenberg Mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/6492
* `WPiOS`: https://github.com/wordpress-mobile/WordPress-iOS/pull/22272

## What?

With this PR, we ensure that font in the editor is changed when users change their accessibility settings on iOS.

## Why?

At the moment, Dynamic Type is not supported in the editor's `RichText` component for iOS. This is because the `refreshFont()` function [here](https://github.com/WordPress/gutenberg/blob/89dbd659f47b25bba62b5cd99c4e14729265d506/packages/react-native-aztec/ios/RNTAztecView/RCTAztecView.swift#L691-L700) is 'undoing' the scaling that takes place elsewhere in Aztec's codebase.

## How?

We ensure scaling is applied to the new font when it's set [here](https://github.com/WordPress/gutenberg/blob/60e287f84be9e416dc6705c71af1e23c0e18c4ea/packages/react-native-aztec/ios/RNTAztecView/RCTAztecView.swift#L654-L660) within the `applyFontConstraints()` function.

## Testing Instructions

> An installable build to test these changes can be found [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/22272).

- Navigate to the **Accessibility** section in the **Settings** app. 
- Select **Display & Text Size** then **Larger Accessibility Sizes** to on.
- Select a larger accessibility size.
- Next, open the WordPress app and select a site with a block-based theme.
- Navigate to the editor in the app.
- Either add a heading block or begin typing into a paragraph block.
- Select the block's cog/gear icon to open its settings and select a larger font size.
- Experiment with typing more text and adding more heading or paragraph blocks.
- Verify that the font has been scaled in accordance with your device and block settings.

## Screenshots or screencast

<details>
 <summary>Before and after ⤵️</summary>

| Before  | After |
| ------------- | ------------- |
| <img src="https://github.com/WordPress/gutenberg/assets/2998162/7068feaa-3fba-433b-b58f-a4a321682f99" width="100%"> | <img src="https://github.com/WordPress/gutenberg/assets/2998162/d84e38a4-be0d-4d54-9e44-34fe709e5b49" width="100%"> |
</details>

<details>
 <summary>Android comparison ⤵️</summary>

| Android  | iOS |
| ------------- | ------------- |
| <img src="https://github.com/WordPress/gutenberg/assets/2998162/844b2642-cf35-4081-804f-d2cb3a0c2a29" width="100%"> | <img src="https://github.com/WordPress/gutenberg/assets/2998162/d84e38a4-be0d-4d54-9e44-34fe709e5b49" width="100%"> |
</details>

SiobhyB committed:
Ensure font is scaled as expected

SiobhyB committed:
Revert accidental new line

SiobhyB committed:
Refactor code for readability/conciseness

SiobhyB committed:
Remove duplicate and incorrect 'let' declaration

SiobhyB commented:
@geriux, @twstokes, I know you have some upcoming AFK, and I do too, so don't expect that this will be prioritised. Just wanted to get it on your radar for when you're back!

twstokes commented on packages/react-native-aztec/ios/RNTAztecView/RCTAztecView.swift:
```diff
@@ -651,12 +651,9 @@ class RCTAztecView: Aztec.TextView {
-        } else {
-            newFontSize = baseFont.pointSize
-        }
+        newFontSize = fontMetrics.scaledValue(for: fontSize ?? baseFont.pointSize)
```
This makes sense to me @SiobhyB!

One optional formatting nit. Since this is now a single line we could update it to:

```swift
let newFontSize = fontMetrics.scaledValue(for: fontSize ?? baseFont.pointSize)
```

SiobhyB replied:
Thank you, Tanner! I updated in https://github.com/WordPress/gutenberg/pull/57339/commits/d98f0b762c0446ae92bac005d9c650352e79e829.

twstokes reviewed (approved):
Worked great for me @SiobhyB! 🚀 

As a future enhancement it would be nice if the editor responded to Dynamic Type changes without needing to be closed and reopened. That's not nearly as important, though, as supporting it in the first place.

SiobhyB committed:
Refactor newFontSize init for clarity/conciseness

A response to feedback noted here: https://github.com/WordPress/gutenberg/pull/57339#discussion_r1435279169

SiobhyB commented:
@twstokes, thank you so much for the review! 🙌  

> As a future enhancement it would be nice if the editor responded to Dynamic Type changes without needing to be closed and reopened. That's not nearly as important, though, as supporting it in the first place.

This makes sense, I've created a new issue at https://github.com/WordPress/gutenberg/issues/57354 listing some of the remaining tasks left to make Dynamic Type support better overall. I've made a note to shape a pitch around accessibility fixes in the New Year which would include this. 🙇‍♀️

geriux commented on packages/react-native-aztec/ios/RNTAztecView/RCTAztecView.swift:
```diff
@@ -650,13 +650,9 @@ class RCTAztecView: Aztec.TextView {
-        } else {
-            newFontSize = baseFont.pointSize
-        }
+        let newFontSize = fontMetrics.scaledValue(for: fontSize ?? baseFont.pointSize)
```
TIL about `scaledValue` 👏
